generic.go

```diff
@@ -161,13 +195,48 @@ func (g *Generic) Run(ctx context.Context) {
 		stage.Start(g.worker)
 	}
 
+	// Stage: start servers
+	stage = stgr.NextStage()
+	ctx, cancel := context.WithCancel(ctx)
+
+	for _, srv := range g.Servers {
+		stage.StartWithContext(func(metricsCtx context.Context) {
+			defer cancel() // if srv fails to start it signals the whole program that it should shut down
+			err := srv.Server.Run(metricsCtx)
+			g.logger.Sugar().Errorf("Server errored out with %s", err)
+		})
+	}
+
 	<-ctx.Done()
 }
 
+func addMiddleware(requestCount *prometheus.CounterVec, requestTime prometheus.Histogram) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			res := chimw.NewWrapResponseWriter(w, r.ProtoMajor)
+			t0 := time.Now()
+			next.ServeHTTP(w, r)
+			t1 := time.Now()
+			tn := t1.Sub(t0)
+
+			requestCount.WithLabelValues(r.URL.Path, r.Method, string(res.Status())).Inc()
+			requestTime.Observe(tn.Seconds())
+		})
+	}
+}
+
 type Holder struct {
 	Cntrlr             Interface
 	ZapNameField       ZapNameField
 	ReadyForWork       <-chan struct{}
 	objectProcessCount prometheus.Counter
 	objectProcessTime  prometheus.Histogram
 }
+
+type ServerHolder struct {
+	Server       Server
+	ZapNameField ZapNameField
+	ReadyForWork <-chan struct{}
+	requestCount *prometheus.CounterVec
```

Comment:
Other ones are not pointers. Why this one is different?

Reply:
prometheus.CounterVec is a struct and Histogram is a interface - they are all actually pointers after running NewHistogram, etc.

Actually I think prometheus defines pointer receiver too for CounterVec